### ISSUE TITLE
Reactive health indicators are ignored in MVC applications

### ISSUE THREAD

**Issue by SirWayne** [spring-projects/spring-boot#18805]
The reactive always returns the http status code 200, even if the health check is down. The response body contains the correct status code.

Response
```
< 200 OK OK
< Date: [Tue, 29 Oct 2019 16:42:00 GMT]
< Content-Type: [application/json]

{"body":{"status":"DOWN"},"status":503}
```

**Comment by philwebb** [comment]
@SirWayne What version of Spring Boot are you using? Do you have a sample application that shows the problem?

**Comment by SirWayne** [comment]
Thanks for the quick reply.

Okay the problem is a little bit more complicated.
We use the WebFlux API with Mono and Flux, but run in a servlet context. With the latest update (2.2.0.RELEASE), the reactive health check classes are no longer registered because the entire structure has changed.
What i have tried now is to create the auto-config. I think i miss a bean. I am currently working on a project.

```
@Configuration( proxyBeanMethods = false )
@ConditionalOnAvailableEndpoint( endpoint = HealthEndpoint.class )
@EnableConfigurationProperties( { HealthEndpointProperties.class, HealthIndicatorProperties.class } )
@Import( { LegacyHealthEndpointAdaptersConfiguration.class, LegacyHealthEndpointCompatibilityConfiguration.class,
               HealthEndpointConfiguration.class, ReactiveHealthEndpointConfiguration.class } )
@EnableAutoConfiguration( exclude = { HealthEndpointAutoConfiguration.class } )
public class ServletReactiveHealthEndpointConfiguration {

   public ServletReactiveHealthEndpointConfiguration(
         final HealthEndpointProperties healthEndpointProperties,
         final HealthIndicatorProperties healthIndicatorProperties ) {
      healthEndpointProperties.getStatus().getOrder().addAll( healthIndicatorProperties.getOrder() );
      healthEndpointProperties.getStatus().getHttpMapping().putAll( healthIndicatorProperties.getHttpMapping() );
   }

   @Bean
   @ConditionalOnMissingBean
   ReactiveHealthEndpointWebExtension reactiveHealthEndpointWebExtension(
         final ReactiveHealthContributorRegistry reactiveHealthContributorRegistry,
         final HealthEndpointGroups groups ) {
      return new ReactiveHealthEndpointWebExtension( reactiveHealthContributorRegistry, groups );
   }
```

**Comment by SirWayne** [comment]
@philwebb here is a small test project with test
https://github.com/SirWayne/demo

**Comment by philwebb** [comment]
Thanks for the sample @SirWayne. I think I've located and fixed the bug in time for 2.2.1.

Your `ServletReactiveHealthEndpointConfiguration` is a little unusual and because it uses `@ConditionalOnMissingBean` annotations it should really only be used as a true auto-configuration. If you `@ComponentScan` or `@Import` it you're likely to get into trouble. Hopefully the fix means that you can at least make some more progress.

**Comment by SirWayne** [comment]
@philwebb thanks :)
